tests/misc/test_logger.cpp

```diff
@@ -0,0 +1,224 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2024 Tenstorrent Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+#include <filesystem>
+#include <fstream>
+#include <string>
+
+#include "common/logger_.hpp"
+#include "common/timestamp.hpp"
+
+using namespace tt::umd::util;
+
+class LoggerTest : public ::testing::Test {
+protected:
+    std::filesystem::path temp_dir;
+    std::filesystem::path log_file;
+
+    void SetUp() override {
+        // A bit of a hack:
+        tt::umd::logger::detail::is_initialized.store(false);
+
+        std::string tmpl = (std::filesystem::temp_directory_path() / "logger_test_XXXXXX").string();
+        int fd = mkstemp(tmpl.data());
+        if (fd == -1) {
+            throw std::runtime_error("Failed to create temporary file");
+        }
+        close(fd);
+        log_file = tmpl;
+        temp_dir = log_file.parent_path();
+    }
+
+    void TearDown() override {
+        std::filesystem::remove_all(log_file);
+        spdlog::shutdown();
+    }
+
+    // Helper to read entire file content
+    std::string read_log_file() {
+        std::ifstream file(log_file);
+        std::stringstream buf;
+        buf << file.rdbuf();
+        return buf.str();
+    }
+};
+
+TEST_F(LoggerTest, BasicLogging) {
+    // Initialize logger with our test configuration
+    tt::umd::logger::Options options;
+    options.log_to_stderr = false;
+    options.filename = log_file.string();
```

Comment:
Does this still print out to stdout?

Reply:
Maybe add a comment that this is printing to a file only. Or add printing to both.